### PR TITLE
It makes no sense to hide the cursor when there is no cursor listed

### PR DESCRIPTION
**Pull request by LocalJoost**
It makes no sense to hide the cursor when there is no cursor listed for a given state. It's better just retain the last one. Otherwise you are obliged to add a cursor for EVERY possible state.

### CODE REVIEW

File: Assets/HoloToolkit/Input/Scripts/Cursor/ObjectCursor.cs [hunk 1]
```diff
@@ -51,21 +51,19 @@ public override void OnCursorStateChange(CursorStateEnum state)
 
             if (state != CursorStateEnum.Contextual)
             {
-                // Hide all children first
-                for(int i = 0; i < ParentTransform.childCount; i++)
+                var newActive = CursorStateData.FirstOrDefault(p => p.CursorState == state);
```
Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Input/Scripts/Cursor/ObjectCursor.cs]: Nit: please use explicit variable declaration.

Comment (LocalJoost) [inline Assets/HoloToolkit/Input/Scripts/Cursor/ObjectCursor.cs]: Oh. I almost never do that. Thank ReSharper for learning me that. Do you want me to change it, or are you just going to merge?

Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Input/Scripts/Cursor/ObjectCursor.cs]: It'd be easier if you did.

File: Assets/HoloToolkit/Input/Scripts/Cursor/ObjectCursor.cs [hunk 2]
```diff
@@ -51,21 +51,19 @@ public override void OnCursorStateChange(CursorStateEnum state)
...
                 {
-                    if (CursorStateData[i].CursorState == state)
-                    {
-                        CursorStateData[i].CursorObject.SetActive(true);
-                    }
+                    oldActive.CursorObject.SetActive(false);
                 }
-            }
+
+                newActive.CursorObject.SetActive(true);
```
Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Input/Scripts/Cursor/ObjectCursor.cs]: It also looks like you're missing a brace to close the outer if statement.

File: Assets/HoloToolkit/Input/Scripts/Cursor/ObjectCursor.cs [hunk 3]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
 using System;
+using System.Linq;
```
Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Input/Scripts/Cursor/ObjectCursor.cs]: https://developer.microsoft.com/en-us/windows/holographic/performance_recommendations_for_unity
> Do not use LINQ, as it causes heavy allocations.
> Do not use lambdas, as they cause allocations.

File: Assets/HoloToolkit/Input/Scripts/Cursor/ObjectCursor.cs [hunk 4]
```diff
@@ -48,23 +54,35 @@ protected override void OnEnable()
...
-                for (int i = 0; i < CursorStateData.Length; i++)
+                if (newActive.Name == null)
                 {
-                    if (CursorStateData[i].CursorState == state)
+                    return;
+                }
+
+                for(int i = 0; i < CursorStateData.Length; i++)
+                {
+                    ObjectCursorDatum c = CursorStateData[i];
```
Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Input/Scripts/Cursor/ObjectCursor.cs]: Nit: please use a variable name longer than one character.

File: Assets/HoloToolkit/Input/Scripts/Cursor/ObjectCursor.cs [hunk 5]
```diff
@@ -48,23 +54,35 @@ protected override void OnEnable()
         public override void OnCursorStateChange(CursorStateEnum state)
         {
             base.OnCursorStateChange(state);
-
             if (state != CursorStateEnum.Contextual)
             {
-                // Hide all children first
```
Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Input/Scripts/Cursor/ObjectCursor.cs]: If we're removing comments, should we put new ones explaining the new behavior?

**Pull request merged**